lumicks/pylake/scan.py

```diff
@@ -91,10 +91,137 @@ def _plot(self, image, frame=1, **kwargs):
+            image_handle.set_data(image)
+
         plt.xlabel(f"{axis_label[axes[0]['axis']]} ($\\mu m$)")
         plt.ylabel(f"{axis_label[axes[1]['axis']]} ($\\mu m$)")
         if self.num_frames == 1:
             plt.title(self.name)
         else:
             plt.title(f"{self.name} [frame {frame}/{self.num_frames}]")
+
+        return image_handle
+
+    def _export_movie(self, plot_type, file_name, start_frame, end_frame, fps, **kwargs):
+        """Export a movie of a particular scan plot
+
+        Parameters
+        ----------
+        plot_type : str
+            One of the plot types "red", "green", "blue" or "rgb".
+        file_name : str
+            File name to export to.
+        start_frame : int
+            First frame in exported movie.
+        end_frame : int
+            Last frame in exported movie.
+        fps : int
+            Frame rate.
+        **kwargs
+            Forwarded to :func:`matplotlib.pyplot.imshow`."""
+        from matplotlib import animation, rcParams
+        import matplotlib.pyplot as plt
+
+        metadata = dict(title=self.name)
+        if "ffmpeg" in animation.writers:
+            writer = animation.writers['ffmpeg'](fps=fps, metadata=metadata)
+        elif "pillow" in animation.writers:
+            writer = animation.writers['pillow'](fps=fps, metadata=metadata)
+        else:
+            raise RuntimeError("You need either ffmpeg or pillow installed to export movies.")
+
+        start_frame = start_frame if start_frame else 1
+        end_frame = end_frame if end_frame else self.num_frames
+        end_frame += 1
+
+        # This prevents font anti-aliasing from being disabled on the notebook backends (at the cost of not having
+        # transparent regions outside the axes part of figures).
+        rcParams['figure.facecolor'] = "White"
+
+        fig = plt.gcf()
+        plot_func = getattr(self, f"plot_{plot_type}")
+        image_handle = plot_func(frame=start_frame, **kwargs)
+
+        def plot(num):
+            plot_func(frame=start_frame + num, image_handle=image_handle, **kwargs)
+            return plt.gca().get_children()
+
+        line_ani = animation.FuncAnimation(fig, plot, end_frame - start_frame, interval=1, blit=True)
+        line_ani.save(file_name, writer=writer)
+        plt.close(fig)
+
+    def export_movie_rgb(self, file_name, start_frame=None, end_frame=None, fps=15, **kwargs):
+        """Export multi-frame scan as movie.
+
+        Parameters
+        ----------
+        file_name : str
+            File name to export to.
+        start_frame : int
+            Initial frame.
+        end_frame : int
+            Last frame.
+        fps : int
+            Frames per second.
+        **kwargs
+            Forwarded to :func:`matplotlib.pyplot.imshow`.
+        """
+        self._export_movie("rgb", file_name, start_frame, end_frame, fps, **kwargs)
+
+    def export_movie_red(self, file_name, start_frame=None, end_frame=None, fps=15, **kwargs):
```

Comment:
Yeah, I agree. This API setup is going to get out of hand at some point. Will think about it.

Reply:
```py
def export_video(self, file_name, channels="rgb", ...):
    ...
```
```py
scan.export_video("scan.gif")
scan.export_video("scan.gif", "r")
scan.export_video("scan.gif", "b")
scan.export_video("scan.gif", "br")
scan.export_video("scan.gif", "bgr")
```
This could be an option but the question is what to do with arbitrary color ordering?

Reply:
I mean, initially, we could just constrain it to the types we already have in plot (i.e. "red", "green", "blue" and "rgb").
I'm not sure whether there are any applications for arbitrary orders or combinations of colors at this point. Otherwise this PR will turn into a complete refactor of `scan` where we also break backward compatibility.

I would say "br" is undefined since you don't know in which channel you want them.

If we are to go with that layout, I'd always make it a tuple of three.
e.g.
"r00"  // Currently "Red"
"0g0" // Currently "Green"
"rg0" // Currently not possible
"rrr"  // Grayscale "Red"
"rgb" // Just "rgb"

That would give users complete control over what is mapped to which channel exactly. But this would require reworking of `Scan` and `Kymo` as well.

Reply:
Oh, no, whatever happens, make it a separate PR. I just wanted to discuss it briefly here.